public/shared/unique.ts

```diff
@@ -0,0 +1,14 @@
+export function unique<TItem, TKey>(list: TItem[], uniqueBy: (list: TItem) => TKey) {
```

Comment:
Might want to constrain `TKey` to `string | number | symbol` otherwise, if people return objects from this function it wont work as expected